src/cli.ts

```diff
@@ -48,30 +50,46 @@ console.log()
 
 const OPENAI_API_KEY =
   process.env.OPENAI_API_KEY ||
-  ((await text({
-    message: dedent`
-      Please provide your OpenAI API key. 
-      
-      To skip this message, set ${chalk.bold('OPENAI_API_KEY')} env variable, and run again. 
-      
-      You can do it in three ways:
-      - by creating an ${chalk.bold('.env.local')} file (make sure to ${chalk.bold('.gitignore')} it)
-        ${chalk.gray(`\`\`\`
-          OPENAI_API_KEY=<your-key>
-          \`\`\`
-        `)}
-      - by passing it inline:
-        ${chalk.gray(`\`\`\`
-          OPENAI_API_KEY=<your-key> npx cali
-          \`\`\`
-        `)}
-      - by setting it as an env variable in your shell (e.g. in ~/.zshrc or ~/.bashrc):
-        ${chalk.gray(`\`\`\`
-          export OPENAI_API_KEY=<your-key>
-          \`\`\`
-        `)}
-    `,
-  })) as string)
+  (await (async () => {
+    let apiKey: string | symbol
+    do {
+      apiKey = await text({
+        message: dedent`
+          ${chalk.bold('Please provide your OpenAI API key.')}
+
+          To skip this message, set ${chalk.bold('OPENAI_API_KEY')} env variable, and run again. 
+          
+          You can do it in three ways:
+          - by creating an ${chalk.bold('.env.local')} file (make sure to ${chalk.bold('.gitignore')} it)
+            ${chalk.gray(`\`\`\`
+              OPENAI_API_KEY=<your-key>
+              \`\`\`
+            `)}
+          - by passing it inline:
+            ${chalk.gray(`\`\`\`
+              OPENAI_API_KEY=<your-key> npx cali
+              \`\`\`
+            `)}
+          - by setting it as an env variable in your shell (e.g. in ~/.zshrc or ~/.bashrc):
+            ${chalk.gray(`\`\`\`
+              export OPENAI_API_KEY=<your-key>
+              \`\`\`
+            `)},
+          `,
+      })
+    } while (typeof apiKey !== 'string')
+
+    const save = await confirm({
+      message: 'Do you want to save it for future runs in `.env.local`?',
+    })
+
+    if (save) {
+      execSync(`echo "OPENAI_API_KEY=${apiKey}" >> .env.local`)
+      execSync(`echo ".env.local" >> .gitignore`)
+    }
```

Comment:
I wonder if saving to each project is a good idea, some tools introduce their own config file for example `ai-cli` created `~/.airc.json`: https://github.com/callstack/ai-cli

I think most of the time user is going to use the same key between every project, wdyt?

Reply:
Yeah maybe let’s write to global config, that’s what I did in my project haha

Reply:
I did change this locally, but I am not sure what is the best way to save into global configuration, so I guess I am going to merge it for now "as is" and we can always re-iterate later!